File: modules/core/src/main/java/kr/entree/enderwand/command/sender/Sender.java
package kr.entree.enderwand.command.sender;

/**
 * Created by JunHyung Lim on 2020-02-04
 */
public interface Sender {
    void sendMessage(Object message);

    default void sendError(Object message) {
        sendMessage(message);
    }

    boolean hasPermission(Object node);

    boolean isOp();
}

File: modules/core-kotlin/src/main/kotlin/enderwand/command/sender/Sender.kt
package enderwand.command.sender

/**
 * Created by JunHyung Lim on 2019-12-18
 */
fun CharSequence.sendTo(sender: Sender) {
    sender.sendMessage(toString())
}

interface Sender {
    fun sendMessage(message: Any)

    fun sendError(errorMessage: Any)

    fun hasPermission(node: String): Boolean

    fun isOp(): Boolean
}